test.nix

```diff
@@ -0,0 +1,54 @@
+{ nixpkgs ? import ./nixpkgs.nix
+, pkgs ? import nixpkgs {}
+}:
+let
+  python = pkgs.python3.withPackages(ps: with ps; [kafka-python]);
+  make-test = import "${nixpkgs}/nixos/tests/make-test-python.nix";
+  test = make-test {
+    name = "kafka-slurp-test";
+    nodes = {
+      slurp = {
+        networking.enableIPv6 = false;
+        environment.systemPackages = [ pkgs.kcat (python) ];
+        services.zookeeper.enable = true;
+        services.apache-kafka.enable = true;
+        systemd.services.slurp = {
+          script = ''
+            ${./kafka-slurp.py} --brokers localhost:9092 --topic testtopic --data-dir /root/out --record-limit 10 --read-timeout 1
+          '';
+          path = [
+            pkgs.lzma
+            python
+          ];
+        };
+      };
+    };
+    skipLint = true;
+    testScript = ''
+      start_all()
+      slurp.wait_for_unit("zookeeper.service")
+      slurp.wait_for_unit("apache-kafka.service")
+      slurp.copy_from_host( "${./kafka-slurp.py}", "/kafka-slurp.py")
+      slurp.copy_from_host( "${./shell.nix}", "/shell.nix")
+      slurp.wait_until_succeeds(
+          "${pkgs.apacheKafka}/bin/kafka-topics.sh --create "
+          + "--zookeeper localhost:2181 --partitions 1 "
+          + "--replication-factor 1 --topic testtopic"
+      )
+      slurp.wait_until_succeeds(
+          "echo testmessage | ${pkgs.kcat}/bin/kcat -b localhost:9092 -t testtopic"
+      )
+      
+      slurp.execute(
+        "systemctl start slurp"
+      )      
+   
+      slurp.execute(
+        "sleep 1"
```

Comment:
wait_until_succeeds (some test to verify the files are output correctly as expected)